### PR TITLE
Sticker manage api CRUD

### PR DESCRIPTION
**Pull request by sadohpv**

### CODE REVIEW

File: libs/components/src/lib/components/ClanSettings/SettingSticker/ModalEditSticker.tsx [hunk 1]
```diff
@@ -1,28 +1,39 @@
+import { createSticker, selectCurrentChannelId, selectCurrentClanId, updateSticker, useAppDispatch } from "@mezon/store";
+import { handleUploadFile, useMezon } from "@mezon/transport";
 import { Button, Icons, InputField } from "@mezon/ui";
-import { ChangeEvent, useState } from "react";
-import { ISticker } from ".";
+// import { createSticker, settingClanStickerActions } from "libs/store/src/lib/settingSticker/settingSticker.slice";
```
Comment (sontung-nccvn) [inline libs/components/src/lib/components/ClanSettings/SettingSticker/ModalEditSticker.tsx]: remove comment

File: libs/components/src/lib/components/ClanSettings/SettingSticker/ModalEditSticker.tsx [hunk 2]
```diff
@@ -34,6 +45,60 @@ const ModalSticker = ({ editSticker, handleCloseModal }: ModalEditStickerProps)
...
+          source: attachment.url,
+        };
+        await dispatch(createSticker({ request: request, clanId: currentClanId }));
+      })
+      handleCloseModal();
+    } else {
+      handleCloseModal();
+      return;
+    }
+  }
```
Comment (sontung-nccvn) [inline libs/components/src/lib/components/ClanSettings/SettingSticker/ModalEditSticker.tsx]: check conditions for early return

File: libs/components/src/lib/components/ClanSettings/SettingSticker/ModalEditSticker.tsx [hunk 3]
```diff
@@ -34,6 +45,60 @@ const ModalSticker = ({ editSticker, handleCloseModal }: ModalEditStickerProps)
...
+  const handleUpadteSticker = async () => {
+    const stickerChange: MezonUpdateClanStickerByIdBody = {
+      source: editSticker?.source?.replace('stickers', 'sticker'),
+      category: editSticker?.category,
+      shortname: editingSticker.shortname
+    }
+    await dispatch(updateSticker({ stickerId: editSticker?.id ?? '', request: stickerChange }))
```
Comment (sontung-nccvn) [inline libs/components/src/lib/components/ClanSettings/SettingSticker/ModalEditSticker.tsx]: do we need to dispatch updateSticker action if editSticker?.id is nullish ?

File: libs/components/src/lib/components/ClanSettings/SettingSticker/SettingStickerItem.tsx [hunk 4]
```diff
@@ -1,31 +1,39 @@
...
 import { Icons } from "../../../components";
 
+import { deleteSticker, selectMemberClanByUserId, useAppDispatch } from "@mezon/store";
+import { useSelector } from "react-redux";
 
 type SettingEmojiListProps = {
-  deleteSticker: (sticker: ISticker) => void;
-  updateSticker: (sticker: ISticker) => void;
-  sticker: ISticker
+  // deleteSticker: (sticker: ApiClanSticker) => void;
```
Comment (sontung-nccvn) [inline libs/components/src/lib/components/ClanSettings/SettingSticker/SettingStickerItem.tsx]: comment

File: libs/components/src/lib/components/ClanSettings/SettingSticker/SettingStickerItem.tsx [hunk 5]
```diff
@@ -1,31 +1,39 @@
...
-const SettingStickerItem = ({ sticker, deleteSticker, updateSticker }: SettingEmojiListProps) => {
+const SettingStickerItem = ({ sticker, updateSticker }: SettingEmojiListProps) => {
+  const dataAuthor = useSelector(selectMemberClanByUserId(sticker.creator_id ?? ''));
+  const dispatch = useAppDispatch();
 
   const handleUpdateSticker = () => {
     updateSticker(sticker);
   }
+  const handleDeleteSticker = async () => {
+    await dispatch(deleteSticker(sticker.id ?? ''));
```
Comment (sontung-nccvn) [inline libs/components/src/lib/components/ClanSettings/SettingSticker/SettingStickerItem.tsx]: check id before invoking function

File: libs/components/src/lib/components/ClanSettings/SettingSticker/ModalEditSticker.tsx (hunk 1 above)
Comment (sadohpv) [reply libs/components/src/lib/components/ClanSettings/SettingSticker/ModalEditSticker.tsx]: Đã fix ạ

File: libs/components/src/lib/components/ClanSettings/SettingSticker/ModalEditSticker.tsx (hunk 2 above)
Comment (sadohpv) [reply libs/components/src/lib/components/ClanSettings/SettingSticker/ModalEditSticker.tsx]: Đã fix ạ

File: libs/components/src/lib/components/ClanSettings/SettingSticker/ModalEditSticker.tsx (hunk 3 above)
Comment (sadohpv) [reply libs/components/src/lib/components/ClanSettings/SettingSticker/ModalEditSticker.tsx]: Đã fix ạ

File: libs/components/src/lib/components/ClanSettings/SettingSticker/SettingStickerItem.tsx (hunk 4 above)
Comment (sadohpv) [reply libs/components/src/lib/components/ClanSettings/SettingSticker/SettingStickerItem.tsx]: Đã fix ạ

File: libs/components/src/lib/components/ClanSettings/SettingSticker/SettingStickerItem.tsx (hunk 5 above)
Comment (sadohpv) [reply libs/components/src/lib/components/ClanSettings/SettingSticker/SettingStickerItem.tsx]: Đã fix ạ

**Pull request merged**